scripts/imgtool/keys/ecdsa.py

```diff
@@ -191,3 +209,84 @@ def sign(self, payload):
             return sig
         else:
             return sig
+
+
+class ECDSA384P1Public(ECDSAPublicKey):
+    """
+    Wrapper around an ECDSA (p384) public key.
+    """
+    def __init__(self, key):
+        super().__init__(key)
+        self.key = key
+
+    def shortname(self):
+        return "ecdsap384"
+
+    def sig_type(self):
+        return "ECDSA384_SHA384"
+
+    def sig_tlv(self):
+        return "ECDSASIG"
```

Comment:
That would introduce a new curve specific TLV, which we are trying to avoid by introducing a new TLV type for all the ECDSA Signature variants, and removing the current ones. The ECDSA224, and ECDSA256 are kept in for backwards compatibility reasons.